packages/experience-builder-sdk/src/hooks/useDetectEditorMode.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useRef, useState } from 'react';
+import {
+  doesMismatchMessageSchema,
+  sendMessage,
+  tryParseMessage,
+} from '@contentful/experience-builder-core';
+import { INCOMING_EVENTS, OUTGOING_EVENTS } from '@contentful/experience-builder-core/constants';
+
+export const useDetectEditorMode = () => {
+  //Assume we are in editor mode initially if we are in iframe
+  const [isEditorMode, setIsEditorMode] = useState(inIframe());
+  const receivedMessage = useRef(false);
+
+  useEffect(() => {
+    const onMessage = (event: MessageEvent) => {
+      if (doesMismatchMessageSchema(event)) {
+        return;
+      }
+      const eventData = tryParseMessage(event);
+
+      if (eventData.eventType === INCOMING_EVENTS.RequestEditorMode) {
+        setIsEditorMode(true);
+        receivedMessage.current = true;
+        if (typeof window !== 'undefined') {
```

Comment:
[nitpick]
We could move this check upward to line 15 and make it a guard clause as it must be defined for a working editor mode:
```
useEffect(() => {
  if (typeof window === 'undefined') return;
```